src/server/req_delete.c

```diff
@@ -785,6 +805,10 @@ req_deletejob2(struct batch_request *preq, job *pjob)
 
 					kill((pid_t) pwtold->wt_event, SIGTERM);
 					set_job_substate(pjob, JOB_SUBSTATE_ABORT);
+					if (preq->rq_type == PBS_BATCH_DeleteJobList) {
+						/* let the caller know that the deljoblist request needs to be suspended */
+						suspenddelete = true;
```

Comment:
Why can't we use a simple return value here, instead of a global variable?

Reply:
i cannot change the function prototype of req_deletejob2 since it gets passed as a callback to dup_br_for_subjob

Reply:
but we can change the definition of dup_br_for_subjob() itself - it is barely called from 7-8 places and not all of them need a return value anyway. I just feel that a global like this is quite dangerous

Reply:
dup_br_for_subjob takes multiple callback functions as input, not just req_deletejob2. If I change its function pointer argument's prototype, then I'll also have to change the prototype of req_signaljob2, req_runjob2 and req_rerunjob2. But I do see your point and I don't like this either, so let me change it.